pkg/xds/client/helpers.go

```diff
@@ -0,0 +1,33 @@
+package client
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+
+	corepb "github.com/cilium/proxy/go/envoy/config/core/v3"
+)
+
+func Node(projectNum int64, mesh string, locality *corepb.Locality, clientID, clientUserAgent string) *corepb.Node {
+	nodeId := fmt.Sprintf("projects/%d/networks/mesh:%s/nodes/%s", projectNum, mesh, clientID)
+	return &corepb.Node{
+		Id:            nodeId,
+		UserAgentName: "xDS client test",
+		Locality:      locality,
+	}
+}
+
+func LocalityFromZone(zone string) *corepb.Locality {
+	if len(zone) < 3 {
+		panic("zone is too short")
```

Comment:
We generally do not want panic calls in production code, there must be a less destructive way to deal with errors than crashing the whole agent.